### DISCUSSION TITLE
Set rated power

### DISCUSSION THREAD

**Discussion by lrmw** [s-allius/tsun-gen3-proxy#92] [Q&A]
Hi, thanks for this awesome solution - I'm using it to get data from my TSUN MS2000 into Home Assistant and it's working great!

Currently there seems to be no supported and working way to increase the rated power limit from 600 to the now legal 800 watts in Germany. I see that your proxy supports reading the rated power via ModBus - is there any way to also set the value in the inverter?

Thanks!

Edit: haha I should have read the latest version announcement, seems like this IS supported through MQTT. Will try setting it through HA today

**Comment by lrmw** [comment]
I tried publishing the following: `mosquitto_pub -h home -t "tsun/proxy/rated_load" -m "800"` 

It doesn't seem to have any effect though, even after restarting the inverter. The config values for my instance are

```
# home-assistant
ha.auto_conf_prefix = 'homeassistant'       # MQTT prefix for subscribing for homeassistant status updates
ha.discovery_prefix = 'homeassistant'       # MQTT prefix for discovery topic 
ha.entity_prefix    = 'tsun'                # MQTT topic prefix for publishing inverter values
ha.proxy_node_id    = 'proxy'               # MQTT node id, for the proxy_node_id
ha.proxy_unique_id  = 'P170000000000001'    # MQTT unique id, to identify a proxy instance
```


EDIT: `mosquitto_pub -h home -t "tsun/inverter/rated_load" -m "800"` seems to do something, at least now I can see `INFO | mqtt | MODBUS via MQTT: tsun/inverter/rated_load = 800'` in the proxy log.

**Reply by lrmw** [reply]
It seems my inverter model won't accept the MODBUS change, after setting it to 800 it continues to report (and produce) only 600 watts. The equipment model is reported as "TSOL-MS2000(600)" (firmware is at LSW5BLE_17_02B0_1.05) so I'm guessing the firmware has the 600 watts limit baked in and I will need to wait for a firmware update :/

**Reply by Cardes** [reply]
I'm unsure if you are setting the rated load the way its meant to be set, can you please check if you used the correct values from [this table](https://github.com/s-allius/tsun-gen3-proxy/wiki/MQTT-topics#inverter-rated-load)to set the rated load?

**Comment by s-allius** [comment]
Hello milancpp,
glad that the project is helping you.

I only send Modbus commands to the inverter once it has sent all its data. I did this to be sure that it is really ready.
You can easily recognize this, because then the `proxy.log`file regularly reports MODBUS data, like this:
```log
'2024-06-14 05:54:00  INFO | data | [inv_1/] MODBUS: inverter.Version : V4.0.10'
'2024-06-14 05:54:00  INFO | data | [inv_1/] MODBUS: grid.Output_Power : 4.2W'
'2024-06-14 05:55:00  INFO | data | [inv_1/] MODBUS: grid.Output_Power : 6.5W'
```
So maybe you have to wait a short time before sending the request (max. 300 sec after the inverter connection is established).

I'm sure that your inverter also supports the required MODBUS command. If it doesn't work, please search in the trace.log file for modbus and post here the related packet dumps. (only the message ID 4510 and 1510 are needed)
